virtool/samples/utils.py

```diff
@@ -57,14 +57,18 @@ def calculate_workflow_tags(analyses: list) -> dict:
     }
 
 
-async def check_labels(labels, postgres_db):
-    non_existent_labels = list()
-    for label_id in labels:
-        async with AsyncSession(postgres_db) as session:
-            if (await session.execute(select(Label).filter_by(id=label_id))).first() is None:
-                non_existent_labels.append(str(label_id))
-
-    return non_existent_labels
+async def check_labels(pg, labels: list) -> list:
+    """"
+    Check for existence of labels given in sample creation request
+
+    :param pg: PostgreSQL database connection object
+    :param labels: list of labels given in the sample creation request
+    :return: a list containing any labels given in the request that aren't in the Labels db, if any
+    """
+    async with AsyncSession(pg) as session:
+        results = await session.execute(select(Label.id).filter(Label.id.in_(labels)))
+
+        return [str(label) for label in labels if label not in set(results.all())]
```

Comment:
Since IDs are integers, this shouldn't be necessary.

Reply:
I believe this was done in order to format a response since `join()` does not work with integers.
https://github.com/virtool/virtool/blob/8ed2beb9000c24800ba5752fefaf39ec2ab60353/virtool/samples/api.py#L240-L241

I could simply have the response return the list itself? e.g. `Labels do not exist: [1, 2, 3]`

Reply:
They are integers. You should not be changing their types so they work with `join()`. You can convert them for the purposes of the error response:

```py
f"Labels do not exist: {', '.join(str(label_id) for label_id in non_existent_labels)}"
```